var/spack/repos/builtin/packages/py-devito/package.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyDevito(PythonPackage):
+    """Devito is a Python package to implement optimized stencil computation.
+
+    (e.g., finite differences, image processing, machine learning) from high-level
+    symbolic problem definitions. Devito builds on SymPy and employs automated code
+    generation and just-in-time compilation to execute optimized computational kernels
+    on several computer platforms, including CPUs, GPUs, and clusters thereof.
+    """
+
+    homepage = "https://www.devitoproject.org/"
+    pypi     = "devito/devito-4.6.2.tar.gz"
+
+    version('4.6.2', sha256='39c2210a192ad69953b4f8d93440ffd72b07d739c4fe2290e2b182adfb7e143f')
+
+    variant('mpi', default=False, description='Enable MPI support')
+    variant('matplotlib', default=False, description='Enable matplolib support')
+    variant('pandas', default=False, description='Enable pandas support')
+
+    depends_on('py-pip@9.0.1:', type='build')
+    depends_on('py-versioneer', type='build')
+    depends_on('py-setuptools', type='build')
+
+    depends_on('py-numpy@1.16:', type=('build', 'run'))
+    depends_on('py-sympy@1.7:1.9', type=('build', 'run'))
+    depends_on('py-scipy', type=('build', 'run'))
+    depends_on('py-flake8@2.1.0:', type=('build', 'run'))
+    depends_on('py-nbval', type=('build', 'run'))
+    depends_on('py-cached-property', type=('build', 'run'))
+    depends_on('py-psutil@5.1.0:5.99', type=('build', 'run'))
+    depends_on('py-py-cpuinfo@8:', type=('build', 'run'))
+    depends_on('py-cgen@2020.1:', type=('build', 'run'))
+    depends_on('py-codepy@2019.1:', type=('build', 'run'))
+    depends_on('py-click@:8.99', type=('build', 'run'))
+    depends_on('py-codecov', type=('build', 'run'))
+    depends_on('py-multidict', type=('build', 'run'))
+    depends_on('py-anytree@2.4.3:2.8', type=('build', 'run'))
+    depends_on('py-pyrevolve@2.1.3:', type=('build', 'run'))
+    depends_on('py-distributed@2022.2:', type=('build', 'run'))
+    depends_on('py-pytest@3.6:7.99', type=('build', 'run'))
+    depends_on('py-pytest-runner', type=('build', 'run'))
+    depends_on('py-pytest-cov', type=('build', 'run'))
+
+    # requirements-mpi.tct
+    depends_on('py-mpi4py@:4.0', type=('build', 'run'), when='+mpi')
+    depends_on('py-ipyparallel@:8.4', type=('build', 'run'), when='+mpi')
+
+    # requirements-optional.txt
+    depends_on('py-matplotlib', type=('build', 'run'), when='+matplotlib')
+    depends_on('py-pandas', type=('build', 'run'), when='+pandas')
+
+    depends_on('mpi', type=('build', 'run'))
+
+    depends_on('intel-parallel-studio', type='run', when='%intel@:2021.1.1')
+    depends_on('intel-oneapi-compilers', type='run', when='%intel@2021.1.2:')
```

Comment:
Compilers aren't currently modeled as dependencies. You can build the Intel compilers in Spack, but you don't add them as dependencies to packages (yet). I would remove these lines.

Reply:
I actually need `icc` during run time. `benchmarks/user/benchmark.py` will compile stuff on the fly while running.